broker/value_added/value_added.py

```diff
@@ -0,0 +1,211 @@
+            err_msg = ("Inconsistent zeropoint values in the epochs of alert {}."
+                        "Cannot continue with classification.").format(oid)
+            assert False, err_msg
+
+        # Set trigger date. fix this.
+        photflag[np.where(flux==np.max(flux))[0][0]] = 6144
+
+        # skip classification if don't have g passband
+        if 'g' not in passband: continue
+        # if 'r' not in passband: continue # this hasn't been a problem yet
+
+        # MW dust extinction
+        # fix this. ZTF docs say ra, dec are in J2000 [deg]
+        coo = SkyCoord(ra, dec, frame='icrs', unit='deg')
+        dust = IrsaDust.get_query_table(coo, section='ebv')
+        mwebv = dust['ext SandF mean'][0]
+
+        # classify for each xmatch host galaxy
+        xm_indicies = oid_map[oid]
+        for xm in xm_indicies:
+            xmatch = xmatch_list[xm]
+            xobjId = xmatch['xobjId']
+            xcatalog = xmatch['xcatalog']
+
+            # skip classification if xobject is not a galaxy
+            if xmatch['sgscore'] > 0.999: continue # ->1 implies star. fix this, high threshold
+            # skip classification if xobject redshift == -1
+            redshift = xmatch['redshift']
+            if redshift < 0: continue
+
+            # get unique alert-hostgal id
+            # this ID is currently *only* used for RAPID *input*, it can be anything
+            cxid = alert_xobj_id([oid, cid, cand_mjd, xcatalog, xobjId ])
+
+            # Collect all the info
+            light_curves.append((np.asarray(mjd), np.asarray(flux), np.asarray(fluxerr), \
+                                np.asarray(passband), np.asarray(photflag), \
+                                ra, dec, cxid, redshift, mwebv))
+
+            cx_dicts[cxid] = {   'objectId': oid,
+                                'candid': cid,
+                                'xobjId': xobjId,
+                                'xcatalog': xcatalog,
+                                'redshift': redshift,
+                                'cand_mjd': cand_mjd
+                            }
+
+    return light_curves, cx_dicts
+
+
+def alert_xobj_id(data):
+    """ Does encoding and decoding of an ID unique to each
+        alert + cross matched object.
+
+        Used as objid input to RAPID.
+
+    Args:
+        data:   (list)  [objectId, alertId, cand_mjd, xcatalog, xobjectId]
+                        as strings or compatible type.
+                        Returns the combined alert_xobj_id
+
+                (str)   alert_xobj_id
+                        Returns list of strings of parsed Id's
+                        [objectId, alertId, cand_mjd, xcatalog, xobjectId]
+
+    Returns:
+        If data is a list of strings, returns the combined alert_xobj_id.
+        If data is a string, returns list of strings of parsed Id's
+    """
+
+    if type(data) == list:
+        assert len(data)==5, 'alert_xobj_id() received invalid data list.'
```

Comment:
Use `raise` not `assert` and `isinstance` instead of `type`.

```
    if isinstance(data, list):
        if len(data)!=5:
            raise ValueError('alert_xobj_id() received invalid data list.')
        
        return '-'.join([str(d) for d in data])

    elif isinstance(data, str):
        return data.split('-')

    else:
        raise ValueError('alert_xobj_id() received invalid data type.')
```